packages/cli/src/commands/doctor/checkInstallation.js

```diff
@@ -0,0 +1,22 @@
+import semver from 'semver';
+import commandExists from 'command-exists';
+
+const PACKAGE_MANAGERS = {
+  YARN: 'YARN',
+  NPM: 'NPM',
+};
+
+const isSoftwareInstalled = async command => {
+  try {
+    await commandExists(command);
+
+    return true;
+  } catch (_ignored) {
+    return false;
+  }
+};
+
+const doesSoftwareNeedToBeFixed = ({version, versionRange}) =>
+  version === 'Not Found' || !semver.satisfies(version, versionRange);
```

Comment:
Note that semver can throw here if for some reason one of the versions are invalid. e.g. for me, I'm getting the string "Using globally installed version of Yarn"

<img width="914" alt="Screenshot 2019-08-07 11 33 17" src="https://user-images.githubusercontent.com/2440089/62616346-28af5680-b907-11e9-8d29-08e0501e153e.png">

We'll at least need to try/catch here, but in the catch we may want to log a quality error and consider crashing because if we don't get a semver by this point that the individual issue needs to diagnose why they don't have a semver

Reply:
How did you install yarn? I want to reproduce that locally.

Reply:
Probably `npm install -g yarn`